client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -35,13 +35,38 @@ class QueryComponent extends Component {
       showConfirmDeleteModal: false,
       showConfirmOverwriteModal: false,
       newQueryName: "",
-      isOverlayResultsChecked: false
+      isOverlayResultsChecked: false,
+      multiline: false,
+      multilineHolder: false,
+      shiftPressed: false
     };
   }
 
   componentDidMount() {
     this.setState({ queries: settingsService.queries });
     eventService.subscribeEnvironmentChange(this.clearAfterEnvironmentChange);
+    document.addEventListener("keyup", this.upFunction, false);
+    document.addEventListener("keydown", this.downFunction, false);
+    document.addEventListener("click", this.onFocusLost, false);
+  }
+
+  upFunction = event => {
+    const shiftPressed = event.keyCode === 16;
+    const enterPressed = event.keyCode === 13;
+    if (this.state.shiftPressed) {
+      if (enterPressed) {
+        this.setState({ multiline: true, multilineHolder: true});
+        this.queryField.focus();
+      }
+    }
+    this.setState({ shiftPressed });
+  }
+
+  downFunction = event => {
+    const shiftPressed = event.keyCode === 16;
+    if (shiftPressed) {
+      this.setState({ shiftPressed });
+    }
```

Comment:
Can we use the shiftKey property on the keyboard event rather than manually keeping track of the shift key state? E.g. https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent